pkg/tests/internal/pulcheck/pulcheck.go

```diff
@@ -29,6 +29,20 @@ import (
 	"gotest.tools/assert"
 )
 
+func resNeedsUpdate(res *schema.Resource) bool {
```

Comment:
What does this supposed to compute?

Reply:
This is to determine if we should add an update - InternalValidate fails for schemas which don't need an update but have it.

Reply:
The `for` loop doesn't do anything here. It only has conditional `continue` statements. If we don't need to check the value of fields, we should remove the `for` loop and just return `res.UpdateContext == nil`. Otherwise we need to make sure that `resNeedsUpdate` is under test for whatever functionality it supports.